### PR TITLE
WEB3-464 - Gas price for front page retrieved from RPC method

### PR DESCRIPTION
**Pull request by davehorizen**
With this PR the gas price displayed on the eon-explorer front page will be retrieved directly from the RPC url calling the `eth_gasPrice` method and not retrieved.
A new module called `GasPrice` was added and it call directly the method on the RPC path calling it in a synchronous way, if something goes wrong during the http call no value will be displayed on the main page.
The already existing module `GasPriceOracle` will supply the blockscout endpoints as before. 
Moreover a bit of refactor was done on the front end modules to remove the _oracle_ references.

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div class="dashboard-banner-chart-legend-item gas-price-from_rpc">
+    <span class="dashboard-banner-chart-legend-label">
+    <%= gettext "Gas tracker" %>
+    </span>
+    <div class="dashboard-banner-chart-legend-value-container">
+        <% gas_prices_from_rpc = gas_price() %>
+        <%= if gas_prices_from_rpc do %>
```
[the remark this answers is not in the document]
Comment (yurikoinaba) [reply apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex:7]: It's against MVC pattern to retrieve the data in the template, as was originally done in `gas_price_oracle_leget_item.html.eex`.

The logic for determining `gas_price` value should be in the [controller](https://github.com/HorizenLabs/eon-explorer/blob/edde39742642513388a479d28e3d9c28166aa627/apps/block_scout_web/lib/block_scout_web/controllers/chain_controller.ex), i.e. determin gas_price = from RPC | from env var | nil

And then in this template file, 
```
<%= if @gas_price do %>
...display the gas price...
<% end %>
```

[the remark this answers is not in the document]
Comment (davehorizen) [reply apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex]: method called moved to the controller and `gas_price_oracle_leget_item.html.eex` refactored

File: apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <span class="dashboard-banner-chart-legend-value inline">
+            <div class="d-flex flex-row" style="height: 20px; line-height: 20px;">
+                <div class="flex-column"><%= "#{gas_prices_from_rpc}" <> " " %><%= gettext "Gwei" %></div>
+            </div>
+        </span>
+        <% else %>
+        <%= if @gas_price do %>
```
[the remark this answers is not in the document]
Comment (davehorizen) [reply apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex:13]: Yes there is the environment variable `GAS_PRICE` read at runtime by the _block scout web_ application. 
I set its default value at 24, did a test returning an error from the rpc endpoint and this default value is displayed on the main page.

File: apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <span class="dashboard-banner-chart-legend-value inline ml-2">
+            <span class="gas-price-icon">
+                <%= render BlockScoutWeb.IconsView, "_gas_price_icon.html" %>
+            </span>
+            <%= @gas_price <> " " %>
+            <%= gettext "Gwei" %>
+        </span>
+        <% end %>
```
Comment (yurikoinaba) [inline apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex:21]: This can be removed

Comment (davehorizen) [reply apps/block_scout_web/lib/block_scout_web/templates/chain/gas_price_legend_item.html.eex]: removed

**Review by yurikoinaba — APPROVED** [review]

**Pull request merged**